realm/realm-library/src/androidTest/java/io/realm/rule/TemporaryRealmFolder.java

```diff
@@ -0,0 +1,66 @@
+package io.realm.rule;
+
+import org.junit.rules.TemporaryFolder;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
+
+/**
+ * Rule that creates {@link RealmConfiguration } in a temporary directory and delete Realms created with those
```

Comment:
Agree with @kneth. Here's another version which has some other grammatical fixes. Feel free to use the below as inspiration or use as is. :)

/**
 * Rule that creates the {@link RealmConfiguration } in a temporary directory and deletes the Realm created with that
 * configuration once the test finishes. Be sure to close all Realm instances before finishing the test. Otherwise
 * {@link Realm#deleteRealm(RealmConfiguration)} will throw an exception in the {@link #after()} method.
 * The temp directory will be deleted regardless if the {@link Realm#deleteRealm(RealmConfiguration)} fails or not.
 */